cmd/createorupdate/update.go

```diff
@@ -43,6 +48,42 @@ func update(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.Plugin)
 	return nil
 }
 
+// needUpdate returns a map with the exact operations that need to be
+// executed during an upgrade. Only scale is supported for now.
+func needUpdate(cs, oldCs *api.OpenShiftManagedCluster) map[string]bool {
+	counts := make(map[string]int)
+	for _, agent := range cs.Properties.AgentPoolProfiles {
+		counts[agent.Name] = agent.Count
+	}
+
+	bm := make(map[string]bool)
+	for _, agent := range oldCs.Properties.AgentPoolProfiles {
+		if counts[agent.Name] != agent.Count {
+			bm["scale"] = true
+		}
+	}
+
+	return bm
+}
+
+func waitForReady(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.Plugin, vmc compute.VirtualMachineScaleSetVMsClient) error {
```

Comment:
Should probably cross-check with scale set count

Reply:
if you have a custom hack which actions scales, it should be able to detect new vms created and just call WaitForReady() on them.  I don't think we shouldn't call it on all of them in case one is broken.
similarly it should have the code to drain vms that will be deleted.  So I think this will need to be refactored a little.

Reply:
>  similarly it should have the code to drain vms that will be deleted. So I think this will need to be refactored a little.

In case of scale down, drain needs to run before we apply the ARM template, as we discussed yesterday, right?

Reply:
After moving `deploy` in `update` this made sense.